### PR TITLE
Documentation for Release/v0.3.0

### PR DESCRIPTION
**Pull request by bburns632**
This PR updates documentation in preparation for version 0.3.0 release on github and CRAN.  I will update and add the website changes from #147 once is available.  

This PR also adds a `NEWS.md` file to `pkgnet`'s set of documentation.  Please review this file in particular for completeness.

 With this PR, here is `pkgnet`'s set of documentation: 
1. `README.md` -  to share `pkgnet`'s purpose and functionality to new users. 
2. `NEWS.md` - to share changes from previous versions with current users
3. `pkgdown` website - to accomplish both of the above in a cohesive website. 
4. `cran-comments.md` - to chronicle communications with CRAN around submissions.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Version 0.3.0
+
+## NEW FEATURES
+* `InheritanceReporter`, a reporter to surface the R6 or S4 relationships within a package.
```
Comment (jayqi) [inline NEWS.md]: Should be:
> `InheritanceReporter`, a reporter for R6, Reference, and S4 class inheritance relationships within a package.

File: NEWS.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Version 0.3.0
+
+## NEW FEATURES
+* `InheritanceReporter`, a reporter to surface the R6 or S4 relationships within a package.  
+* Dropdown menu in graph visualizations for node selection.
+
+## CHANGES
+* Edge direction reversed to align with UML class diagram convention.
```
Comment (jayqi) [inline NEWS.md]: I think this would be more clear if you specify which it is now.

> Now: if A depends on B, then A->B.

File: NEWS.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+## NEW FEATURES
+* `InheritanceReporter`, a reporter to surface the R6 or S4 relationships within a package.  
+* Dropdown menu in graph visualizations for node selection.
+
+## CHANGES
+* Edge direction reversed to align with UML class diagram convention. 
+* Default node colors are now colorblind accessible. 
+* Function network design utilizes graphopt layout.
+* Testing strategy with subpackages are now CRAN and TRAVIS compatible. 
+* Test package `milne` created for unit testing of `InheritanceReporter`.
```
Comment (jayqi) [inline NEWS.md]: It's also used for testing the R6 method support in `FunctionReporter`. 

> Test package `milne` created for unit testing of object-oriented functionality in `InheritanceReporter` and `FunctionReporter`.

File: NEWS.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Version 0.3.0
+
+## NEW FEATURES
+* `InheritanceReporter`, a reporter to surface the R6 or S4 relationships within a package.  
+* Dropdown menu in graph visualizations for node selection.
+
+## CHANGES
+* Edge direction reversed to align with UML class diagram convention. 
+* Default node colors are now colorblind accessible. 
+* Function network design utilizes graphopt layout.
```
Comment (jayqi) [inline NEWS.md]: > `FunctionReporter` uses graphopt layout by default.

Also, maybe should mention:

> Reporters now support all layouts available in `igraph`. Use `grep("^layout_\\S", getNamespaceExports("igraph"), value = TRUE)` to see valid options.

File: NEWS.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+* Testing strategy with subpackages are now CRAN and TRAVIS compatible. 
+* Test package `milne` created for unit testing of `InheritanceReporter`.
+* Width of html reports adjust to size of screen. 
+
```
Comment (jayqi) [inline NEWS.md]: > Orphaned node clustering was removed in favor of using layout to better handle graphs with many orphaned nodes.

Comment (jayqi) [inline NEWS.md]: > `FunctionReporter` now supports non-exported functions and R6 class methods.

**Comment by bburns632** [comment]
I made all suggested edits.  In the future, feel free to suggest edits & additions directly for this type of `NEWS.md` update.

**Comment by bburns632** [comment]
We need to handle gif in some other means.  4.6 MB tar ball seems excessive. 
image

Here's `lubridate` for perspective.  
image

**Comment by bburns632** [comment]
This change references the gif via the raw.githubusercontent.com path.  Should reduce tar ball size and allow sweet gifs on this repo's readme, `pkgdown` site, and CRAN (probably).  I'll mention the reason for the link in the release notes to CRAN.

**Comment by bburns632** [comment]
Just resubmitted.

**Pull request merged**